From 3c87c890fc3e01cdbd12a3eb86c3c6f5ccdc3273 Mon Sep 17 00:00:00 2001
From: Yong Zhi <yong.zhi@intel.com>
Date: Fri, 13 Apr 2018 15:27:18 -0500
Subject: [PATCH 018/129] media: intel-ipu4: bus: Use updated
 init_iova_domain() API

Message for Open Source:
init_iova_domain() function parameters has changed in kernel 4.15, so
add kernel version checking for it.

Message for Internal:
[Issue/Feature] Refer to the Message for Open Source above.
[Root Cause/Changes] pfn parameter was dropped in init_iova_domain(),
so update code accordingly.

Tracked-On: #JKCCIB-201
Change-Id: Ie37b0ae3163167f798d2825fc528cc449c120488
Signed-off-by: Yong Zhi <yong.zhi@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-bus.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-bus.c b/drivers/media/pci/intel/ipu-bus.c
index 336f3a3..53908e3 100644
--- a/drivers/media/pci/intel/ipu-bus.c
+++ b/drivers/media/pci/intel/ipu-bus.c
@@ -189,11 +189,13 @@ static struct ipu_dma_mapping *alloc_dma_mapping(struct device *dev)
 		return NULL;
 	}
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 0, 0)
-	init_iova_domain(&dmap->iovad,
-#else
+	init_iova_domain(&dmap->iovad, dma_get_mask(dev) >> PAGE_SHIFT);
+#elif LINUX_VERSION_CODE < KERNEL_VERSION(4, 15, 0)
 	init_iova_domain(&dmap->iovad, SZ_4K, 1,
-#endif
 			 dma_get_mask(dev) >> PAGE_SHIFT);
+#else
+	init_iova_domain(&dmap->iovad, SZ_4K, 1);
+#endif
 
 	kref_init(&dmap->ref);
 
-- 
1.9.1

